service/qos/service_level_controller.hh

```diff
@@ -45,6 +46,8 @@ struct service_level {
      bool is_static;
 };
 
+using update_only_effective_cache = bool_class<class update_only_effective_cache_tag>;
```

Comment:
I think it would be better to introduce a type called `update_service_level_cache` and have it set to `yes` by default as the argument to `update_cache`.

Parameters such as `update_only_effective_cache` have less chance to be orthogonal with future parameters due to the "only" part; imagine if you want to add more flags like this that will control reloading of other stuff.